presto-hive/src/main/java/io/prestosql/plugin/hive/procedure/DropStatsProcedure.java

```diff
@@ -86,20 +89,27 @@ public Procedure get()
                 DROP_STATS.bindTo(this));
     }
 
-    public void dropStats(ConnectorSession session, String schema, String table, List<?> partitionValues)
+    public void dropStats(ConnectorSession session, ConnectorAccessControl accessControl, String schema, String table, List<?> partitionValues)
     {
         try (ThreadContextClassLoader ignored = new ThreadContextClassLoader(getClass().getClassLoader())) {
-            doDropStats(session, schema, table, partitionValues);
+            doDropStats(session, accessControl, schema, table, partitionValues);
         }
     }
 
-    private void doDropStats(ConnectorSession session, String schema, String table, List<?> partitionValues)
+    private void doDropStats(ConnectorSession session, ConnectorAccessControl accessControl, String schema, String table, List<?> partitionValues)
     {
         TransactionalMetadata hiveMetadata = hiveMetadataFactory.create();
         HiveTableHandle handle = (HiveTableHandle) hiveMetadata.getTableHandle(session, new SchemaTableName(schema, table));
         if (handle == null) {
             throw new PrestoException(INVALID_PROCEDURE_ARGUMENT, format("Table '%s' does not exist", new SchemaTableName(schema, table)));
         }
+
+        try {
+            accessControl.checkCanDeleteFromTable(null, new SchemaTableName(schema, table));
+        }
+        catch (AccessDeniedException exception) {
+            throw new AccessDeniedException(format("Cannot drop stats for table %s.%s", schema, table));
```

Comment:
Same comment. The original error message will be good if we have an explicit access check for statistics.